src/main/java/com/hillel/items_exchange/security/jwt/JwtTokenProvider.java

```diff
@@ -60,20 +63,25 @@ private String getUsername(String token) {
 
     public String resolveToken(HttpServletRequest req) {
         String bearerToken = req.getHeader(AUTHORIZATION_HEADER_NAME);
-        if (bearerToken != null && bearerToken.startsWith(BEARER_PREFIX)) {
-            return bearerToken.substring(7);
+        if(bearerToken != null) {
+            if(bearerToken.startsWith(BEARER_PREFIX)) {
+                return bearerToken.substring(7);
+            } else {
+                log.error("Unauthorized: {}", messageSource.getMessage("token.not.start.with.bearer", null, Locale.US));
+                req.setAttribute("detailedError", messageSource.getMessage("token.not.start.with.bearer", null, Locale.US));
+                return null;
```

Comment:
Would you mind to return empty string instead of null and updated the method that invokes current one?

Reply:
Performed